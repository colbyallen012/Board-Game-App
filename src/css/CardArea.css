.cardContainer {
    background-color: #DCE6EC;
    display: grid;
    grid-template-columns: repeat(4,1fr);
    grid-column-gap: 10px;
    grid-row-gap: 15px;
    justify-items: center;
    padding-top: 15px;
    width: 100%;
}

