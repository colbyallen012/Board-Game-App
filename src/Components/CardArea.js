import React, {Component} from 'react';
import App from '../App.js';
import NavBar from './NavBar.js'
import Game from './Game.js'
import Location from './Location.js'
import '../css/CardArea.css'

class CardArea extends Component {
  constructor(props) {
    super(props);
    this.state= {

    }
  }

  render() {
    if (this.props.locationData) {
    let gamesCards =
     this.props.gamesData.map((game) => {
      return <Game
      name={game.name}
      minPlayers={game.minPlayers}
      maxPlayers={game.maxPlayers}
      weight={game.weight}
      timeToPlay={game.timeToPlay}
      type={game.type}
      features={game.features}
      image={game.image}
      id={game.gameId}
      />
  // let locationCards =
  //    this.props.locationData.map((location) => {
  //     return <Location
  //     name={location.name}
  //     address = {location.address}
  //     website = {location.website}
  //     hasFood = {location.hasFood}
  //     hasDrinks = {location.hasDrinks}
  //     sellsGames = {location.sellsGames}
  //     weekdayOpen = {location.weekdayOpen}
  //     weekdayClose = {location.weekdayClose}
  //     weekendOpen = {location.weekendOpen}
  //     weekendClose = {location.weekendClose}
  //     googleMapsLink = {location.googleMapsLink}
  //     bringYourOwnGame = {location.bringYourOwnGame}
  //     gamesOffered = {location.gamesOffered}/>
    });
    return (
      <div>
        <NavBar NavBar={NavBar}/>
        <section className = 'cardContainer'>
          {gamesCards}
        </section>
      </div>
    )
  } else {
    return (
      <div>
        {this.props.cardArea}
      </div>
      )
    }
  }
}

export default CardArea;
