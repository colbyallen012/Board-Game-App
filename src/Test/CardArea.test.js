import React from 'react';
import ReactDOM from 'react-dom';
import CardArea from '../Components/CardArea';
import Navbar from '../Components/Navbar';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


const favorites = []

const mockGames = [{
name: "Betrayal at House on the Hill",minPlayers: 3,maxPlayers: 6,weight: "medium",
timeToPlay: 60,type: "thematic",features: ["adventure","hidden traitor","dice rolling","cooperative"],
image: "https://i.imgur.com/t12PXuZ.jpg",gameId: 4},
{name: "Boss Monster",minPlayers: 2,maxPlayers: 4,weight: "medium",timeToPlay: 20,type: "strategy",features: ["auction","family"],
image: "https://i.imgur.com/pbutakO.jpg",gameId: 5},
{name: "Camel Up",minPlayers: 2,maxPlayers: 8,weight: "medium",timeToPlay: 30,type: "party",
features: ["dice rolling","family","wagering"],image: "https://i.imgur.com/93iAMYa.jpg",gameId: 6}
]

const mockLocation = [{
    name: "Bardo Coffee House",
    address: "238 South Broadway, Denver, CO 80209",
    website: "http://www.bardocoffee.com",
    hasFood: true,
    hasDrinks: true,
    sellsGames: false,
    weekdayOpen: "6AM",
    weekdayClose: "12AM",
    weekendOpen: "6AM",
    weekendClose: "12AM",
    bringYourOwnGame: true,
    logo: "https://i.imgur.com/8zbmUrg.png",
    gamesOffered: [7,8,15,52,53,54]
}]

const playerFilter = jest.fn() 

const weightFilter = jest.fn()

const gameTypeFilter = jest.fn()

const foodFilter = jest.fn()

const drinkFilter = jest.fn()

const sellerFilter = jest.fn()

const bringGameFilter = jest.fn()

const searchByText = jest.fn()

const toggleFav = jest.fn()

const filterFavorites = jest.fn()

const changeCards = jest.fn() 

const mockEvent = {target: {id:  'Locations' } };

describe('CardArea', () => {
  let wrapper;
  
  beforeEach(() =>{
    wrapper = shallow(
      <CardArea
        gamesData={mockGames}
        locationData={mockLocation}
        favorites= {favorites}
        playerFilter={playerFilter}
        weightFilter={weightFilter}
        gameTypeFilter={gameTypeFilter}
        foodFilter={foodFilter}
        drinkFilter={drinkFilter}
        sellerFilter={sellerFilter}
        bringGameFilter={bringGameFilter}
        searchByText={searchByText}
        toggleFav={toggleFav}
        filterFavorites={filterFavorites}
        />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

   it('should have the proper default state', () => {
    expect(wrapper.state('view')).toEqual('Games');
  });

  // it('should toggle view', () => {
  //   wrapper.setState({view: 'Games'})
  //   expect(wrapper.state('view')).toEqual('Games');
  //   wrapper.find('.location toggle-button').simulate('click', {
  //     preventDefault: () => {}
  //   })
  //   wrapper.instance().changeCards();
  //   expect(wrapper.state('view')).toEqual('Locations')
  // });

});
