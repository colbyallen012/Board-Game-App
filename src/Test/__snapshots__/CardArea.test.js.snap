// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardArea should match the snapshot with all data passed in correctly 1`] = `
<div>
  <NavBar
    bringGameFilter={[MockFunction]}
    changeCards={[Function]}
    drinkFilter={[MockFunction]}
    foodFilter={[MockFunction]}
    gameTypeFilter={[MockFunction]}
    games={
      Array [
        Object {
          "features": Array [
            "adventure",
            "hidden traitor",
            "dice rolling",
            "cooperative",
          ],
          "gameId": 4,
          "image": "https://i.imgur.com/t12PXuZ.jpg",
          "maxPlayers": 6,
          "minPlayers": 3,
          "name": "Betrayal at House on the Hill",
          "timeToPlay": 60,
          "type": "thematic",
          "weight": "medium",
        },
        Object {
          "features": Array [
            "auction",
            "family",
          ],
          "gameId": 5,
          "image": "https://i.imgur.com/pbutakO.jpg",
          "maxPlayers": 4,
          "minPlayers": 2,
          "name": "Boss Monster",
          "timeToPlay": 20,
          "type": "strategy",
          "weight": "medium",
        },
        Object {
          "features": Array [
            "dice rolling",
            "family",
            "wagering",
          ],
          "gameId": 6,
          "image": "https://i.imgur.com/93iAMYa.jpg",
          "maxPlayers": 8,
          "minPlayers": 2,
          "name": "Camel Up",
          "timeToPlay": 30,
          "type": "party",
          "weight": "medium",
        },
      ]
    }
    playerFilter={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    sellerFilter={[MockFunction]}
    status="Games"
    weightFilter={[MockFunction]}
  />
  <section
    className="cardContainer"
  >
    <Game
      features={
        Array [
          "adventure",
          "hidden traitor",
          "dice rolling",
          "cooperative",
        ]
      }
      id={4}
      image="https://i.imgur.com/t12PXuZ.jpg"
      maxPlayers={6}
      minPlayers={3}
      name="Betrayal at House on the Hill"
      timeToPlay={60}
      type="thematic"
      weight="medium"
    />
    <Game
      features={
        Array [
          "auction",
          "family",
        ]
      }
      id={5}
      image="https://i.imgur.com/pbutakO.jpg"
      maxPlayers={4}
      minPlayers={2}
      name="Boss Monster"
      timeToPlay={20}
      type="strategy"
      weight="medium"
    />
    <Game
      features={
        Array [
          "dice rolling",
          "family",
          "wagering",
        ]
      }
      id={6}
      image="https://i.imgur.com/93iAMYa.jpg"
      maxPlayers={8}
      minPlayers={2}
      name="Camel Up"
      timeToPlay={30}
      type="party"
      weight="medium"
    />
  </section>
</div>
`;
